yt/frontends/amrvac/data_structures.py

```diff
@@ -245,56 +245,50 @@ def _set_code_unit_attributes(self):
-            unit_pressure = ((2.0 + 3.0*He_abundance) *
-                             unit_numberdensity * boltzmann_constant_cgs  * unit_temperature).to('dyn*cm**-2')
-            unit_velocity = (np.sqrt(unit_pressure / unit_density)).to('cm*s**-1')
+            # if nothing is specified, use the default AMRVAC normalisations
+            length_override = (1, 'cm')
+            numberdensity_override = (1, 'cm**-3')
+            temperature_override = (1, 'K')
+            velocity_override = (0, 'cm*s**-1')
+
+        # Create YT quantities
+        length_unit = self.quan(length_override[0], length_override[1])
+        numberdensity_unit = self.quan(numberdensity_override[0], numberdensity_override[1])
+        temperature_unit = self.quan(temperature_override[0], temperature_override[1])
+        velocity_unit = self.quan(velocity_override[0], velocity_override[1])
+
+        He_abundance = 0.1  # hardcoded parameter in AMRVAC
+        density_unit = (1.0 + 4.0*He_abundance) * mass_hydrogen_cgs * numberdensity_unit
+        if velocity_unit == 0:
+            pressure_unit = ((2.0 + 3.0*He_abundance) *
+                             numberdensity_unit * boltzmann_constant_cgs  * temperature_unit).to('dyn*cm**-2')
+            velocity_unit = (np.sqrt(pressure_unit / density_unit)).to('cm*s**-1')
         else:
-            unit_pressure = (unit_density * unit_velocity**2).to('dyn*cm**-2')
-            unit_temperature = (unit_pressure /
-                                ((2.0 + 3.0*He_abundance) * unit_numberdensity * boltzmann_constant_cgs)).to('K')
-        unit_time = unit_length / unit_velocity
-        unit_mass = unit_density * unit_length**3
-        unit_magneticfield = (np.sqrt(4*np.pi * unit_pressure)).to('gauss')
-
-        setdefaultattr(self, "length_unit", unit_length)
-        setdefaultattr(self, "mass_unit", unit_mass)
-        setdefaultattr(self, "time_unit", unit_time)
-
-        setdefaultattr(self, "velocity_unit", unit_velocity)
-        setdefaultattr(self, "density_unit", unit_density)
-        setdefaultattr(self, "numberdensity_unit", unit_numberdensity)
-
-        setdefaultattr(self, "temperature_unit", unit_temperature)
-        setdefaultattr(self, "pressure_unit", unit_pressure)
-        setdefaultattr(self, "magnetic_unit", unit_magneticfield)
+            pressure_unit = (density_unit * velocity_unit**2).to('dyn*cm**-2')
+            temperature_unit = (pressure_unit /
+                                ((2.0 + 3.0*He_abundance) * numberdensity_unit * boltzmann_constant_cgs)).to('K')
+        time_unit = length_unit / velocity_unit
+        mass_unit = density_unit * length_unit**3
+        magneticfield_unit = (np.sqrt(4*np.pi * pressure_unit)).to('gauss')
+
+        # Finally we set the attributes. This extends the attributes coming from Dataset._override_code_units().
+        # Those that are already defined as attributes (i.e. those supplied to override_units when loading) are
+        # not overridden.
```

Comment:
this comment doesn't add anything new (see "note" at the top of this function)

Reply:
True, removed it